Utilities/DavixAdaptor/src/DavixFile.cc

```diff
@@ -0,0 +1,363 @@
+#include "Utilities/DavixAdaptor/interface/DavixFile.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/Utilities/interface/EDMException.h"
+#include "FWCore/Utilities/interface/Exception.h"
+#include <cassert>
+#include <davix.hpp>
+#include <errno.h>
+#include <fcntl.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <vector>
+
+using namespace Davix;
+
+DavixFile::DavixFile(void) {}
+
+DavixFile::DavixFile(const char *name, int flags /* = IOFlags::OpenRead */, int perms /* = 066 */) {
+  open(name, flags, perms);
+}
+
+DavixFile::DavixFile(const std::string &name, int flags /* = IOFlags::OpenRead */,
+                     int perms /* = 066 */) {
+  open(name.c_str(), flags, perms);
+}
+
+DavixFile::~DavixFile(void) {
+  configureDavixLogLevel();
+  close();
+  return;
+}
+
+void DavixFile::close(void) {
+  if (davixPosix && m_fd) {
+    DavixError *err;
+    davixPosix->close(m_fd, &err);
+  }
+  return;
+}
+
+void DavixFile::abort(void) {
+  if (davixPosix && m_fd) {
+    DavixError *err;
+    davixPosix->close(m_fd, &err);
+  }
+  return;
+}
+
+void DavixFile::configureDavixLogLevel() {
+  long logLevel = 0;
+  char *logptr;
+  char *davixDebug = getenv("Davix_Debug");
+  if (davixDebug != NULL) {
+    logLevel = strtol(davixDebug, &logptr, 0);
+    if (errno) {
+      edm::LogWarning("DavixFile") << "Got error while converting "
+                                   << "Davix_Debug env variable to integer. "
+                                      "Will use default log level 0";
+      logLevel = 0;
+    }
+    if (logptr == davixDebug) {
+      edm::LogWarning("DavixFile") << "Failed to convert to integer "
+                                   << "Davix_Debug env variable; Will use default log level 0";
+      logLevel = 0;
+    } else if (*logptr != '\0') {
+      edm::LogWarning("DavixFile") << "Failed to parse extra junk "
+                                   << "from Davix_Debug env variable. Will use default log level 0";
+      logLevel = 0;
+    }
+  }
+  switch (logLevel) {
+  case 0:
+    davix_set_log_level(0);
```

Comment:
I checked and it turns out this is not technically thread safe because the underlying global variable being set is not `atomic`.
https://github.com/cern-it-sdc-id/davix/blob/25976712a9c88277894324f6280f23e5b5955958/src/utils/davix_logger.cpp#L33

In principal I'm not worried about this under intel like machines but could be a problem under other architectures.